src/Microsoft.Azure.WebJobs.Logging/Internal/Utility.cs

```diff
@@ -72,5 +74,80 @@ public static string DecodeBase64(string base64)
             string str = Encoding.UTF8.GetString(bytes);
             return str;
         }
+
+
+        private static async Task SafeCreateAsync(this CloudTable table)
+        {
+            var interval = TimeSpan.FromSeconds(5);
+            var total = TimeSpan.FromSeconds(120);
+
+            while(true)
+            {
+                try
+                {
+                    await table.CreateIfNotExistsAsync();
+                    return; 
+                }
+                catch (StorageException e)
+                {
+                    var code = (HttpStatusCode)e.RequestInformation.HttpStatusCode;
+                    // This can throw 409 if the table is in the process of being deleted.                     
+                    if (code != HttpStatusCode.Conflict)
+                    {
+                        throw;
+                    }
+
+                    if (total.TotalMilliseconds < 0)
+                    {
+                        // timeout. 
+                        throw;
+                    }
+                }
+                await Task.Delay(interval);
+                total = total.Subtract(interval);
+            }
+        }
+
+        // Write table entry. 
+        // If table doesn't exist (such as if it was deleted), then recreate it. 
+        public static async Task SafeWriteAsync(this CloudTable table, TableBatchOperation batch)
```

Comment:
I'd rename this to SafeExecuteAsync since it isn't write specific - it's just executing an arbitrary operation